hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java

```diff
@@ -146,8 +147,15 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         }
 
         // If the fee schedule key is changed then the transaction must also be signed by the new fee schedule key
-        if (op.hasFeeScheduleKey()) {
-            context.requireKeyOrThrow(op.feeScheduleKey(), INVALID_FEE_SCHEDULE_KEY);
+        if (op.hasFeeScheduleKey()
+                && !KeyOneOfType.UNSET.equals(op.feeScheduleKey().key().kind())) {
+            context.requireKeyOrThrow(op.feeScheduleKey(), INVALID_CUSTOM_FEE_SCHEDULE_KEY);
```

Comment:
If modifying feeScheduleKey, don't we also need current feeScheduleKey to sign too?

Reply:
As far as I understand it - no. It's enough to be signed by the admin key and the new feeScheduleKey